modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1769,7 +1770,138 @@ def sort_by(
+                ]
+                overall_quantiles = np.quantile(all_pivots, quantiles)
+                overall_quantiles[0] = np.NINF
+                overall_quantiles[-1] = np.inf
+                return overall_quantiles
+
+            def split_func(df, pivots):
+                """
+                Split the given dataframe into the partitions specified by `pivots`.
+
+                This function takes as input a row-axis partition, as well as the quantiles determined
+                by the `pivot_func` defined above. It then splits the input dataframe into NPartitions.get() * 2
+                dataframes, with the elements in the i-th split belonging to the i-th partition, as determined
+                by the quantiles we're using.
+                """
+                groupby_col = np.digitize(df[columns[0]].squeeze(), pivots)
+                grouped = df.groupby(groupby_col)
+                groups = [
+                    grouped.get_group(i)
+                    if i in grouped.keys
+                    else pandas.DataFrame(columns=df.columns)
+                    for i in range(len(pivots))
+                ]
+                if not ascending:
+                    groups = groups[::-1]
+                return tuple(groups)
+
+            new_partitions = self._partition_mgr_cls.shuffle_partitions(
+                self._partitions, sample_func, pivot_func, split_func
+            )
+            new_partitions = self._partition_mgr_cls.lazy_map_partitions(
+                new_partitions,
+                lambda df: df.sort_values(by=columns, ascending=ascending),
+            )
+            new_axes = [0, 0]
+            new_axes[axis.value] = self._compute_axis_labels(0, new_partitions)
+            new_axes[axis.value ^ 1] = self.axes[axis.value ^ 1]
+            new_lengths = [0, 0]
+            new_lengths = [None, None]
+            new_partitions = self._partition_mgr_cls.rebalance_partitions(
+                new_partitions
+            )
+            return self.__constructor__(
+                new_partitions, *new_axes, *new_lengths, self.dtypes
+            )
+        elif (
+            axis == Axis.ROW_WISE
+            and len(columns) == 1
+            and columns[0] == self.index.names[0]
+        ):
+            new_partitions = self._partition_mgr_cls.map_axis_partitions(
+                axis.value,
+                self._partitions,
+                lambda df: df.sort_index(ascending=ascending),
+            )
+            new_axes = [0, 0]
+            new_axes[axis.value] = self._compute_axis_labels(0, new_partitions).rename(
+                self.axes[axis.value].names[0]
+            )
+            new_axes[axis.value ^ 1] = self.axes[axis.value ^ 1]
+            new_lengths = [0, 0]
+            new_lengths = [None, None]
+            return self.__constructor__(
+                new_partitions, *new_axes, *new_lengths, self.dtypes
+            )
```

Comment:
Will there ever be a case where neither the `if` or the `elif` is hit?

Reply:
Not likely - I imagine we'll route column sorts differently at the query compiler layer.

Reply:
Is this worth adding and else case and throwing an error if the case isn't hit?

Reply:
Not sure - the query compiler checks if we have a multi index and defaults to pandas if so, and the dataframe checks if its a column-wise sort, and routes it to a different method in the query compiler if so, so its really very unlikely the else case will be hit unless someone is specifically writing code at the query_compiler/dataframe level